notebooks/4. Pole-like objects.ipynb

```diff
@@ -0,0 +1,329 @@
+    "\n",
+    "First we try to match all objects in the BGT data with pole-like objects in the point cloud. We look for clusters of points with specific characteristics, such as a minimum height, density, and a maximum radius.\n",
+    "\n",
+    "We start with trees, since the location data for these is most precise (in Amsterdam)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7293838a-be2f-47c8-98fc-0959ce9d8ee9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# All possible parameters:\n",
+    "# params = {'search_pad': 1.5,   # Specify the padding (in m) around the BGT object in which to search for a match in the point cloud.\n",
+    "#           'max_dist': 1.2,     # Maximum distance (in m) between the expected location and the location of a potential match.\n",
+    "#           'voxel_res': 0.2,    # Resolution of voxels used when searching for a match.\n",
+    "#           'seed_height': 1.75, # Height above ground at which to determine object dimensions.\n",
+    "#           'min_height': 2.,    # Minimum hieght for an object to be considered a match.\n",
+    "#           'max_r': 0.5,        # Maximum radius for a pole-like object to be considered a match.\n",
+    "#           'min_points': 500,   # Minimum number of points for a cluster to be considered.\n",
+    "#           'z_min': 0.2,        # Height above ground, above which to search for objects.\n",
+    "#           'z_max': 2.7,        # Height above ground, below which to search for objects.\n",
+    "#           'r_mult': 1.5,       # Multiplier for radius when performing the initial (cylinder-based) labelling.\n",
+    "#           'label_height': 4.}  # Maximum height for initial (cylinder-based) labelling.\n",
+    "\n",
+    "# Object type-specific parameters.\n",
+    "tree_params = {'seed_height': 1.75, 'min_points': 500, 'max_r': 0.5, 'label_height': 5.}\n",
+    "light_params = {'seed_height': 2.25, 'min_points': 400, 'max_r': 0.2, 'label_height': 5.}\n",
+    "sign_params = {'seed_height': 1.75, 'min_points': 200, 'max_r': 0.2, 'min_height': 1.2, 'z_max': 2., 'label_height': 3.}\n",
+    "\n",
+    "# Create fusers for BGT point data.\n",
+    "tree_fuser = fusion.BGTPointFuser(Labels.TREE, bgt_type='boom', bgt_file=bgt_data_file, ahn_reader=ahn_reader, params=tree_params)\n",
+    "light_fuser = fusion.BGTPointFuser(Labels.STREET_LIGHT, bgt_type='lichtmast', bgt_file=bgt_data_file, ahn_reader=ahn_reader, params=light_params)\n",
+    "sign_fuser = fusion.BGTPointFuser(Labels.TRAFFIC_SIGN, bgt_type='verkeersbord', bgt_file=bgt_data_file, ahn_reader=ahn_reader, params=sign_params)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3da03370-80a6-43b2-941a-a9f7db038f86",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Process the point cloud.\n",
+    "mask = (labels != Labels.GROUND) & (labels != Labels.BUILDING) & (labels != Labels.CAR)\n",
```

Comment:
Opzich zou je alles in de pipeline structuur kunnen verwerken, net zoals de Cars notebook doet

Reply:
Goed idee